apis/src/main/java/xyz/bookself/controllers/user/UserController.java

```diff
@@ -62,19 +62,28 @@ public UserController(UserRepository repository, BookListRepository bookListRepo
 
     @PostMapping(value = "/new-user", consumes = {MediaType.APPLICATION_JSON_VALUE})
     public ResponseEntity<User>createNewUser(@RequestBody UserDto userDto){
-        User newUser = new User();
 
-        newUser.setUsername(userDto.getUsername());
-        newUser.setEmail(userDto.getEmail());
-        var passwordHasher = new BCryptPasswordEncoder();
-        String hashedPass = passwordHasher.encode(userDto.getPassword());
-        newUser.setPasswordHash(hashedPass);
-        newUser.setCreated(LocalDate.now());
-        userRepository.save(newUser);
-
-        createNewBookLists(newUser);
+        //check to see if username/email already exists and is non unique.
+        //if it exists, return a 409 error
+        boolean x = userRepository.findUserByUsername(userDto.getUsername()).isPresent();
+        boolean y = userRepository.findUserByEmail(userDto.getEmail()).isPresent();
+        if (!(x || y)) {
```

Comment:
Might be more readable if you add an exists method on the userRepository that checks if there are records with the given email and/or username.

then this could be simplified to `if (!userRepository.exists(userDto.getUsername(), userDto.getEmail()) {`

Reply:
will refactor in another commit if time permits